components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/store/AuthorizationStoreImpl.java

```diff
@@ -423,14 +414,10 @@ public Permission getPermission(String resource, String action) throws Permissio
 
         List<Role> roles = new ArrayList<>();
 
-        RealmService realmService = CarbonSecurityDataHolder.getInstance().getCarbonRealmService();
-
         for (AuthorizationStoreConnector authorizationStoreConnector : authorizationStoreConnectors.values()) {
             roles.addAll(authorizationStoreConnector.getRolesForGroup(groupId, domain.getDomainName())
                     .stream()
-                    .map(roleBuilder -> roleBuilder
-                            .setAuthorizationStore(realmService.getAuthorizationStore())
-                            .build())
+                    .map(Role.RoleBuilder::build)
```

Comment:
Need to add authorizationStoreConnectorId before building the Role. Or else build method will throw an exception.